File: app/src/main/java/com/trayis/simplimvvm/app/SampleApp.java
package com.trayis.simplimvvm.app;

import android.app.Application;
import android.content.Context;

import com.trayis.simplimvvm.utils.SimpliProviderUtil;
import com.trayis.simplimvvm.utils.SimpliResource;
import com.trayis.simplimvvmannotation.generated.SimpliMvvmProviderImpl;

public class SampleApp extends Application implements SimpliResource {

    @Override
    public void onCreate() {
        super.onCreate();

        SimpliProviderUtil.setProvider(SimpliMvvmProviderImpl.getInstance(this));
    }

    public static SampleApp getInstance(Context context) {
        return (SampleApp) context.getApplicationContext();
    }

}

File: app/src/main/java/com/trayis/simplimvvm/app/SampleApp.kt
package com.trayis.simplimvvm.app

import android.app.Application
import android.content.Context
import com.trayis.simplimvvm.utils.SimpliProviderUtil
import com.trayis.simplimvvmannotation.generated.SimpliMvvmProviderImpl

class SampleApp : Application() {

    companion object {
        fun getInstance(context: Context) = context.applicationContext as SampleApp
    }

    override fun onCreate() {
        super.onCreate()

        SimpliProviderUtil.setProvider(SimpliMvvmProviderImpl(this))
    }
}
